File: examples/python/query.py
import json
import os
import urllib.parse
import urllib.request

base_url = os.environ.get("BASE_URL", "http://localhost:1279")
api_key = os.environ.get("API_KEY")

if not api_key:
    raise SystemExit("Please set API_KEY=cp_live_xxx")

params = urllib.parse.urlencode({
    "author": "李白",
    "q": "月",
    "search_in": "content",
    "page": 1,
    "page_size": 5,
})

req = urllib.request.Request(
    f"{base_url}/api/v1/poems/query?{params}",
    headers={"X-API-Key": api_key},
)

with urllib.request.urlopen(req, timeout=10) as resp:
    payload = json.loads(resp.read().decode("utf-8"))

print(json.dumps(payload, ensure_ascii=False, indent=2))

File: examples/javascript/query.mjs
const baseUrl = process.env.BASE_URL || "http://localhost:1279";
const apiKey = process.env.API_KEY;

if (!apiKey) {
  throw new Error("Please set API_KEY=cp_live_xxx");
}

const params = new URLSearchParams({
  author: "李白",
  q: "月",
  search_in: "content",
  page: "1",
  page_size: "5",
});

const res = await fetch(`${baseUrl}/api/v1/poems/query?${params}`, {
  headers: { "X-API-Key": apiKey },
});

if (!res.ok) {
  throw new Error(`HTTP ${res.status}: ${await res.text()}`);
}

console.log(JSON.stringify(await res.json(), null, 2));
